.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,93 @@
+# .github/workflows/pr-preview.yml
+name: Deploy PR previews
+
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - closed
+    paths:
+      - '**/*.md'
+
+concurrency: preview-${{ github.ref }}
+
+jobs:
+  deploy-preview:
+    if: github.event.action == 'opened' || github.event.action == 'reopened'  || github.event.action == 'synchronize'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout master branch
+        uses: actions/checkout@v2
+        with:
+          ref: master
+          fetch-depth: 0
+      - name: Install mdbook
+        run: |
+          cargo install --git https://github.com/montyly/mdBook.git mdbook || true
+      - name: Build artifacts
+        run: mdbook build
+      - name: Cache master build artifacts
+        uses: actions/cache/save@v3
+        with:
+          path: ./book
+          key: master-book
+      - name: Checkout pull request 
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+      - name: Install mdbook
+        run: |
+          cargo install --git https://github.com/montyly/mdBook.git mdbook || true
+      - name: Build artifacts
+        run: mdbook build
+      - name: Deploy preview
+        uses: rossjrw/pr-preview-action@v1
```

Comment:
Do we need this step? I was under the impression that we were building book + book/pr-review, and uploading this